src/main/java/com/google/starfish/servlets/UploadGNoteServlet.java

```diff
@@ -0,0 +1,122 @@
+public class UploadGNoteServlet extends HttpServlet {
+
+  private final String COOKIE_NAME = "SFCookie";
+  private LabelService labelService = new LabelService();
+  
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
+    if(!validateUser(request)) {
+      response.setStatus(HttpServletResponse.SC_FORBIDDEN);
+      return;
+    }
+
+    HttpSession activeSession = request.getSession(false);
+    String userId = (String) activeSession.getAttribute("user_id");
+
+    String title = request.getParameter("title");
+    String school = request.getParameter("school").toLowerCase();
+    String course = request.getParameter("course").toLowerCase();
+    String sourceUrl = request.getParameter("sourceUrl");
+    String pdfSource = request.getParameter("pdfSource");
+
+    DataSource pool = (DataSource) request.getServletContext().getAttribute("my-pool");
+
+    // Insert all labels before inserting note due to foreign key constraint
+    addLabels(pool, school, course);
+    
+    try (Connection conn = pool.getConnection()) {
+      String stmt =
+          "INSERT INTO notes ( "
+              + "author_id,"
+              + "school,"
+              + "course,"
+              + "title,"
+              + "source_url,"
+              + "pdf_source,"
+              + "date_created,"
+              + "num_downloads ) "
+        + "VALUES ( "
+              + "?,"
+              + "?,"
+              + "?,"
+              + "?,"
+              + "?,"
+              + "?,"
+              + "?,"
+              + "0 ); ";
+
+      try (PreparedStatement noteStmt = conn.prepareStatement(stmt, new String[] {"id"})) {
+        noteStmt.setString(1, userId);
+        noteStmt.setString(2, school);
+        noteStmt.setString(3, course);
+        noteStmt.setString(4, title);
+        noteStmt.setString(5, sourceUrl);
+        noteStmt.setString(6, pdfSource);
+        noteStmt.setDate(7, new Date(Calendar.getInstance().getTimeInMillis()));
+        noteStmt.executeUpdate();
+      }
+    } catch (SQLException ex) {
+      response.setStatus(HttpServletResponse.SC_INTERNAL_SERVER_ERROR);
+      response.getWriter().println("INTERNAL SERVER ERROR: " + ex);
+    }    
+  }
+
+  /** Inserts a school and course label to the database */
+  private void addLabels(DataSource pool, String school, String course) {
+    labelService.insertSchoolLabel(pool, school);
+    labelService.insertCourseLabel(pool, course);
+  }
+
+  /**
+   * Validates the user using the request's session and cookies. If there is a
+   * valid user logged in, returns true. Otherwise, returns false.
+   **/
+  private boolean validateUser(HttpServletRequest req) {
+    Cookie[] cookies = req.getCookies();
+    String sessionId = null;
+    for (Cookie cookie : cookies) {
+      if (COOKIE_NAME.equals(cookie.getName())) {
+        sessionId = cookie.getValue();
+        break;
+      }
+    }
```

Comment:
Not a huge deal, but possible todo is to make this a static method of a `ValidateUser` class or something because we're using this exact code in more than one servlet.

Reply:
Yeah that makes sense, maybe we can add this in a future PR.